babel/babel
# drop support for Node.js v4

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #7753
| Patch: Bug Fix?          | n/a
| Major: Breaking Change?  | Yes
| Minor: New Feature?      | n/a
| Tests Added + Pass?      | Yes
| Documentation PR         | n/a
| Any Dependency Changes?  | n/a
| License                  | MIT

- remove version 4 from Travis build matrix
- update environment support doc
- update `CONTRIBUTING.md`
- update `engines` field of root `package.json`

If I need to make changes elsewhere, please let me know.  I didn't see that the `engines` field was present (nor getting copied) into packages' `package.json`; maybe it should be.

(I'll also look to see if the website needs a PR for this.)
